libkbfs/folder_branch_ops.go

```diff
@@ -1206,6 +1206,15 @@ func (fbo *folderBranchOps) initMDLocked(
 		return err
 	}
 
+	fbo.headLock.Lock(lState)
+	defer fbo.headLock.Unlock(lState)
+	// Some other thread got here first, so give up and let it go
+	// before we push anything to the servers.
```

Comment:
This causes headLock to be held for the remainder of the function, which seems bad since it's across network ops.

But we only change the head when we hold both `writerLock` and `headLock`. (see `setHeadLocked`). So I think it suffices to check at the top of this function that `getHead()` is `nil`, and return early otherwise.

Does that sound right?